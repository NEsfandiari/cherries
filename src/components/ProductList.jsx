// Component for mapping out ProductCard component with passed-in product list

import React, { Component } from 'react'
import { Link } from 'gatsby'
import styled from 'styled-components'
import ProductCard from './ProductCard'

const Container = styled.div`
  margin: 0 auto;
  width: 75vw;
  .product-list-card-container {
    display: flex;
    flex-wrap:wrap;
    justify-content: space-between;
  }
  .product-list-header {
   color: #E20031 ;
   font-family: Glacial Indifference;
   font-weight: 700
   font-size: 30px;
   line-height: 39px;
   margin-bottom: 0;
   text-align: left;
   font-style: italic;
  }
  .product-list-subheader{
    font-size: 16px;
  }
  a{
    color: #E20031 ;
  };
  .product-list-subheader{
    margin-left: 30px
  }
`
class ProductList extends Component {
  constructor(props) {
    super(props)
    this.state = {
      shopDropdown: 'sort by',
    }
    this.handleChange = this.handleChange.bind(this)
  }

  handleChange(evt) {
    //TODO: Work on Method
    //PROP METHOD GOES HERE
    this.setState({ [evt.target.name]: evt.target.value })
  }

  render() {
    console.log('what does the state look like', this.state)
    const header = this.props.catalog ? 'Shop All' : 'Fresh Picks'
    const subheader = this.props.catalog ? (
      <form htmlFor="shopDropdown">
        <select
          name="shopDropdown"
          id="shopDropdown"
          onChange={this.handleChange}
          value={this.state.shopDropdown}
        >
          <option value="sortBy">Sort by</option>
          <option value="featured">Featured</option>
          <option value="recentlyAdded">Recently Added</option>
          <option value="rating">Rating</option>
          <option value="priceLowToHigh">Price: Low to High</option>
          <option value="priceHighToLow">Price: High to Low</option>
        </select>
      </form>
    ) : (
      <Link to="/">
        <u>
          <i>
            <small>
              <span className="product-list-subheader">Show all</span>
            </small>
          </i>
        </u>
      </Link>
    )
    let products = this.props.products.map(product => (
      <ProductCard
        className="ProductCard"
        key={product.node.id}
        product={product.node}
      />
    ))

    // Section title and link to be refactored to be dynamic
    return (
      <Container>
        <p className="product-list-header">
          Fresh picks{' '}
          <Link to="/">
            <span className="product-list-subheader">Shop all</span>
          </Link>
        </p>
        <div className="product-list-card-container">{products}</div>
      </Container>
    )
  }
}

export default ProductList
