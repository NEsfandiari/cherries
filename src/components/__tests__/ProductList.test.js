import React from 'react'
import renderer from 'react-test-renderer'
import ProductList from '../molecules/ProductList'

describe('ProductList', () => {
  it('renders correctly', () => {
    const products = [
      {
        node: {
          title: 'Astronaut Baby',
          price: 22.99,
          sku: '24',
          productCopy: {
            content: [
              {
                nodeType: 'paragraph',
                content: [
                  {
                    value: 'Hi',
                    nodeType: 'text',
                  },
                ],
              },
            ],
          },
          details: {
            content: [
              {
                content: [
                  {
                    value: 'Materials: something something something',
                  },
                ],
              },
              {
                content: [
                  {
                    value: 'Closure: something else',
                  },
                ],
              },
              {
                content: [
                  {
                    value: 'Allergy Information: something the most',
                  },
                ],
              },
            ],
          },
          images: [
            {
              file: {
                url:
                  '//images.ctfassets.net/4yejk6rho2rp/60XuYBiY6IIes0aqyAOQIs/e177ec00483171ad7d8a71e6cf8f462f/fashion-woman-dangle-earrings-heartly-space-dream-tong-qu-the-earth-rocket-astronauts-blue-pendants-earrings.jpg_640x640.jpg',
              },
            },
            {
              file: {
                url:
                  '//images.ctfassets.net/4yejk6rho2rp/7hDZI4CKeQa4wAamoU0sCI/111d798be6ef67e2e5a827d00a3a832d/fashion-woman-dangle-earrings-heartly-space-dream-tong-qu-the-earth-rocket-astronauts-blue-pendants-earrings.jpg',
              },
            },
            {
              file: {
                url:
                  '//images.ctfassets.net/4yejk6rho2rp/3ok4nPAB3iosEWIsSQAeUA/a110bde70a54bf93a5aede31bed4b651/fashion-woman-dangle-earrings-heartly-space-dream-tong-qu-the-earth-rocket-astronauts-blue-pendants-earrings.jpg',
              },
            },
          ],
        },
      },
      {
        node: {
          title: 'Geometric Oval Drop',
          price: 18.99,
          sku: '18',
          productCopy: {
            content: [
              {
                nodeType: 'paragraph',
                content: [
                  {
                    value:
                      'Bro ipsum dolor sit amet pow Ski reverse camber, caballerial rail bail skinny endo travel betty ski bum. Schwag dope north shore ride around over the bars. Table top huck crank, caballerial japan air rigid misty chain ring roadie wheels drop huck. Travel switch huckfest ACL wheelie misty derailleur daffy Whistler rig.',
                    nodeType: 'text',
                  },
                ],
              },
            ],
          },
          details: null,
          images: [
            {
              file: {
                url:
                  '//images.ctfassets.net/4yejk6rho2rp/7slN8SXJ0QiKkcYCyGkSyO/9b62e2aefc280bbc818a5f7dbab3a515/Fashion-color-metal-geometry-stud-earrings-female-popular-Classic-Golden-women-stud-earrings-red-rectangle-earrings.jpg',
              },
            },
            {
              file: {
                url:
                  '//images.ctfassets.net/4yejk6rho2rp/5taayuJdde2MYkCyw6I64k/2894909da007fb66428e6db98485c04b/Fashion-color-metal-geometry-stud-earrings-female-popular-Classic-Golden-women-stud-earrings-red-rectangle-earrings.jpg',
              },
            },
            {
              file: {
                url:
                  '//images.ctfassets.net/4yejk6rho2rp/6HdQmSobdeWEKQia88KmOg/4f19626bcd98a0d64a1d6e0a4eadd439/Fashion-color-metal-geometry-stud-earrings-female-popular-Classic-Golden-women-stud-earrings-red-rectangle-earrings.jpg',
              },
            },
          ],
        },
      },
    ]
    const tree = renderer.create(<ProductList products={products} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
