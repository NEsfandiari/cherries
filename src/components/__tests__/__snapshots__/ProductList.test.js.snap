// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList renders correctly 1`] = `
<div
  className="sc-jTzLTM kakAjN"
>
  <p
    className="product-list-header"
  >
    Fresh Picks
    <span
      className="product-list-subheader"
    >
      <a
        href="/"
      >
        <u>
          <i>
            <small>
              <span
                className="product-list-subheader"
              >
                Show all
              </span>
            </small>
          </i>
        </u>
      </a>
    </span>
  </p>
  <div
    className="product-list-card-container"
  >
    <div
      className="sc-ifAKCX hDJblS"
    >
      <a
        className="product-card"
        href="Astronaut-Baby/"
      >
        <img
          alt="earring"
          className="product-card-image"
          src="//images.ctfassets.net/4yejk6rho2rp/60XuYBiY6IIes0aqyAOQIs/e177ec00483171ad7d8a71e6cf8f462f/fashion-woman-dangle-earrings-heartly-space-dream-tong-qu-the-earth-rocket-astronauts-blue-pendants-earrings.jpg_640x640.jpg"
        />
        <p
          className="product-card-name"
        >
          Astronaut Baby
        </p>
        <p
          className="product-card-price"
        >
          $
          22.99
        </p>
      </a>
    </div>
    <div
      className="sc-ifAKCX hDJblS"
    >
      <a
        className="product-card"
        href="Geometric-Oval-Drop/"
      >
        <img
          alt="earring"
          className="product-card-image"
          src="//images.ctfassets.net/4yejk6rho2rp/7slN8SXJ0QiKkcYCyGkSyO/9b62e2aefc280bbc818a5f7dbab3a515/Fashion-color-metal-geometry-stud-earrings-female-popular-Classic-Golden-women-stud-earrings-red-rectangle-earrings.jpg"
        />
        <p
          className="product-card-name"
        >
          Geometric Oval Drop
        </p>
        <p
          className="product-card-price"
        >
          $
          18.99
        </p>
      </a>
    </div>
  </div>
</div>
`;
